backend/capellacollab/projects/toolmodels/modelsources/git/github/interface.py

```diff
@@ -0,0 +1,197 @@
+    git_model: git_models.DatabaseGitModel,
+) -> t.Any:
+    response = requests.get(
+        f"https://api.github.com/repos{project_id}/actions/runs?branch={parse.quote(git_model.revision, safe='')}&per_page=20",
+        timeout=2,
+    )
+
+    if not response.ok:
+        response = requests.get(
+            f"https://api.github.com/repos{project_id}/actions/runs?branch={parse.quote(git_model.revision, safe='')}&per_page=20",
+            headers={
+                "Authorization": f"token {git_model.password}",
+                "X-GitHub-Api-Version": "2022-11-28",
+                "Accept": "application/vnd.github+json",
+            },
+            timeout=2,
+        )
+    response.raise_for_status()
+    return response.json()["workflow_runs"]
+
+
+def get_artifact_from_job_as_content(
+    project_id: str,
+    job_id: str,
+    trusted_path_to_artifact: str,
+    git_model: git_models.DatabaseGitModel,
+    git_instance: settings_git_models.DatabaseGitInstance,
+) -> bytes:
+    return get_artifact_from_job(
+        project_id, job_id, trusted_path_to_artifact, git_model, git_instance
+    ).encode()
+
+
+def get_artifact_from_job_as_json(
+    project_id: str,
+    job_id: str,
+    trusted_path_to_artifact: str,
+    git_model: git_models.DatabaseGitModel,
+    git_instance: settings_git_models.DatabaseGitInstance,
+) -> dict:
+    return json.loads(
+        get_artifact_from_job(
+            project_id,
+            job_id,
+            trusted_path_to_artifact,
+            git_model,
+            git_instance,
+        )
+    )
+
+
+def get_artifact_from_job(
+    project_id: str,
+    job_id: str,
+    trusted_path_to_artifact: str,
+    git_model: git_models.DatabaseGitModel,
+    git_instance: settings_git_models.DatabaseGitInstance,
+) -> str:
+    response = requests.get(
+        f"{git_instance.api_url}/repos{project_id}/actions/runs/{job_id}/artifacts",
+        headers={
+            "Authorization": f"token {git_model.password}",
+            "X-GitHub-Api-Version": "2022-11-28",
+            "Accept": "application/vnd.github+json",
+        },
+        timeout=2,
+    )
+    response.raise_for_status()
+    artifact_id = response.json()["artifacts"][0]["id"]
+
+    if response.json()["artifacts"][0]["expired"] == "true":
+        raise fastapi.HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail={
+                "err_code": "ARTIFACT_EXPIRED",
+                "reason": (
+                    "The latest artifact you are requesting expired. Please rerun your pipline and contact your administrator."
+                ),
+            },
+        )
+    artifact_response = requests.get(
+        f"{git_instance.api_url}/repos{project_id}/actions/artifacts/{artifact_id}/zip",
+        headers={
+            "Authorization": f"token {git_model.password}",
+            "X-GitHub-Api-Version": "2022-11-28",
+            "Accept": "application/vnd.github+json",
```

Comment:
I would prefer to have this as global variable. In a global variable, we could bump the Github version in one place.